src/main/scala/org/apache/spark/sql/SparkExtensions.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 AstroLab Software
+ * Author: Yash Datta
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.LogicalRDD
+import org.apache.spark.sql.functions.lit
+
+object SparkExtensions {
+
+  /**
+   * Optimized Spark SQL equivalent of RDD.zipWithIndex.
```

Comment:
It is optimized in what sense?

Reply:
Avoids the serialization / deserialization from/to df's InternalRow's underlying bytes array <--> GenericRow's underlying JVM objects collection Array[Any])

Reply:
Excellent - could you add that in the documentation?

Reply:
ok